Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -403,37 +400,25 @@ - (void)writeAvatarToDisk:(UIImage *)avatar
     });
 }
 
-- (NSData *)encryptedAvatarData:(NSData *)plainTextData outDigest:(NSData **)outDigest
-{
-    DDLogError(@"TODO: Profile encryption scheme not yet settled.");
-
-    // server accepts up to 14 base64 chars for digest
-    // 14 <= 4 * ceil(n/3)
-    NSUInteger kAvatarDigestByteLength = 9;
-    *outDigest = [Cryptography computeSHA256Digest:plainTextData truncatedToBytes:kAvatarDigestByteLength];
-
-    return plainTextData;
-}
-
 - (void)uploadAvatarToService:(NSData *)avatarData
-                     fileName:(NSString *)fileName // TODO do we need filename?
-                      success:(void (^)(NSString *avatarUrl, NSData *avatarDigest))successBlock
+                      success:(void (^)(NSString *avatarUrl))successBlock
                       failure:(void (^)())failureBlock
 {
     OWSAssert(avatarData.length > 0);
-    OWSAssert(fileName.length > 0);
     OWSAssert(successBlock);
     OWSAssert(failureBlock);
 
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSData *encryptedAvatarDigest;
-        NSData *encryptedAvatarData = [self encryptedAvatarData:avatarData outDigest:&encryptedAvatarDigest];
+        NSData *encryptedAvatarData = [self encryptProfileData:avatarData];
         OWSAssert(encryptedAvatarData.length > 0);
-        OWSAssert(encryptedAvatarDigest.length > 0);
 
         // See: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-UsingHTTPPOST.html
         TSProfileAvatarUploadFormRequest *formRequest = [TSProfileAvatarUploadFormRequest new];
 
+        // TODO: Since this form request causes the server to reset my avatar URL, if the update fails
+        // at some point from here on out, we want the user to understand they probably no longer have
+        // a profile avatar on the server.
```

Comment:
I'm not sure I understand. 

My position was that since it's the `GET v1/profile/form` that effectively removes the avatar, maybe we should do it here (which is strictly earlier than the upload that you're suggesting).